src/Transactions/TransactionLineFields/PeriodField.php

```diff
@@ -24,9 +24,9 @@ public function getPeriod(): ?string
      * @param string $period
      * @return $this
      */
-    public function setPeriod(string $period): self
+    public function setPeriod(?string $period): self
     {
-        if (!preg_match("!\\d{4}/\\d{1,2}!", $period)) {
+        if (!empty($period) && !preg_match("!\\d{4}/\\d{1,2}!", $period)) {
```

Comment:
I prefer actual strict `null` checks, as `empty` can match all kinds of things in PHP (such as `"0"`).